RunTests.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python3
+import os
+import WebpageFunctions
+import argparse
+import json
+
+#Get a set of arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("--test_num", required=True ,help="The test number to run. This is defined by the Test number in tests.json. This argument is required.",type=int,default=1)
+args = parser.parse_args()
+
+# build the comparison script
+ValidationPath = os.getenv("ValidationPath")
+os.chdir(ValidationPath)
+os.system("make")
+
+#Grab environment variables, if not exisiting, replace with default value.
+TRAVIS_PULL_REQUEST = os.getenv("TRAVIS_PULL_REQUEST","false")
+TRAVIS_PULL_REQUEST_TITLE = os.getenv("TRAVIS_PULL_REQUEST_TITLE","")
+TRAVIS_PULL_REQUEST_LINK = os.getenv("TRAVIS_PULL_REQUEST_LINK","")
+TRAVIS_COMMIT = os.getenv("TRAVIS_COMMIT","")
+GITHUBTOKEN = os.getenv("GITHUBTOKEN","")
+
+WebpageFunctions.checkout_validation_webpage_branch(ValidationPath,TRAVIS_PULL_REQUEST,TRAVIS_COMMIT)
+
+ret = 0
+####### Running tests #######
+# Each test has it's own webpage, so lets just build it as we go
+TESTDIR=f"{ValidationPath}/Webpage/{TRAVIS_COMMIT}/{args.test_num}"
+
+TESTWEBPAGE = f"{TESTDIR}/index.html"
+# First make the directory
+os.system(f"mkdir {TESTDIR}")

```

Comment:
You're correct, best practice is usually to use `os.mkdir()` as it will return an appropriate error when a problem occurs vs `os.system("mkdir...")` which will just return a failed exit status with no information on the problem. Will change! 

This goes the same for most uses of `os.system()` calls, if there is an inbuilt function, you should use that instead, so I will check further lines too!